src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php

```diff
@@ -33,10 +34,37 @@ class AssetsInstallCommand extends ContainerAwareCommand
     const METHOD_ABSOLUTE_SYMLINK = 'absolute symlink';
     const METHOD_RELATIVE_SYMLINK = 'relative symlink';
 
+    private $filesystem;
+
     /**
-     * @var Filesystem
+     * @param Filesystem|null $filesystem
      */
-    private $filesystem;
+    public function __construct($filesystem = null)
+    {
+        parent::__construct();
+
+        if (!$filesystem instanceof Filesystem) {
+            @trigger_error(sprintf('Passing a command name as the first argument of "%s" is deprecated since version 3.4 and will be removed in 4.0. If the command was registered by convention, make it a service instead.', __METHOD__), E_USER_DEPRECATED);
+
+            $this->setName(null === $filesystem ? 'assets:install' : $filesystem);
+
+            return;
+        }
+
+        $this->filesystem = $filesystem ?: new Filesystem();
```

Comment:
kinda yes.. in 4.0 it would be `?? ...` and in 3.4 the else may not happen (depends on discussion above :))